modules/dxp/apps/portal-search-similar-results/portal-search-similar-results-web/src/main/java/com/liferay/portal/search/similar/results/web/internal/contributor/asset/publisher/AssetPublisherSimilarResultsContributor.java

```diff
@@ -139,23 +164,53 @@ private Long _getId(AssetEntry assetEntry) {
 				return null;
 			}
 
-			return journalArticle.getId();
+			return journalArticle;
+		}
+		else if (Objects.equals(
+					WikiPage.class.getName(), assetEntry.getClassName())) {
+
+			WikiPage wikiPage =
+				_wikiPageLocalService.fetchWikiPageByUuidAndGroupId(
+					assetEntry.getClassUuid(), assetEntry.getGroupId());
+
+			if (wikiPage == null) {
+				return null;
+			}
+
+			return wikiPage;
+		}
+		else if (Objects.equals(
+					BlogsEntry.class.getName(), assetEntry.getClassName())) {
+
+			BlogsEntry blogsEntry =
+				_blogsEntryLocalService.fetchBlogsEntryByUuidAndGroupId(
+					assetEntry.getClassUuid(), assetEntry.getGroupId());
+
+			if (blogsEntry == null) {
+				return null;
+			}
+
+			return blogsEntry;
```

Comment:
Can we sort these `if` checks alphabetically? 
1. BlogsEntry
2. JournalArticle
3. WikiPage

Reply:
nice catch!